### PR TITLE
Move Streamly.data.Unfold to core

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: src/Streamly/Internal/Data/Stream/IsStream/Unfold.hs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+-- Module      : Streamly.Internal.Data.Stream.IsStream.Unfold
+-- Copyright   : (c) 2017 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+
+module Streamly.Internal.Data.Stream.IsStream.Unfold
+    (fromStream)
+where
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Stream/IsStream/Unfold.hs:11]: A module only for this looks like an overkill. Can it be moved to IsStream.Generate instead?

File: src/Streamly/Internal/Data/Stream/IsStream/Unfold.hs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+-- XXX Using Unfold.fromStreamD seems to be faster (using cross product test
+-- case) than using fromStream even if it is implemented using fromStreamD.
+-- Check if StreamK to StreamD rewrite rules are working correctly when
+-- implementing fromStream using fromStreamD.
+
+-- | Convert a stream into an 'Unfold'. Note that a stream converted to an
+-- 'Unfold' may not be as efficient as an 'Unfold' in some situations.
+--
+-- /Since: 0.8.0/
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Stream/IsStream/Unfold.hs:26]: We will have to change the since notation, as this is new to this module.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]
Need to update the changelog as well.

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/1750

File: src/Streamly/Internal/Data/Stream/IsStream/Generate.hs [hunk 3]
```diff
@@ -662,3 +664,17 @@ fromCallback setCallback = concatM $ do
...
+-- implementing fromStream using fromStreamD.
+
+-- | Convert a stream into an 'Unfold'. Note that a stream converted to an
+-- 'Unfold' may not be as efficient as an 'Unfold' in some situations.
+--
+-- /Since: 0.9.0/
+--
+{-# INLINE_NORMAL unfoldFromStream #-}
+unfoldFromStream :: (IsStream t, Applicative m) => Unfold m (t m a) a
+unfoldFromStream = Unfold.lmap IsStream.toStream Unfold.fromStreamK
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Stream/IsStream/Generate.hs:680]: We can call this as `readStream` to keep it shorter.

**Review by harendra-kumar — COMMENTED** [review]
squash the commits to ready for merge.

**Review by rnjtranjan — COMMENTED** [review]
Looks fine.

**Pull request merged**